src/main/webapp/app/entities/share/share-visualization.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useRef, useState } from 'react';
+import {
+    ActionButton,
+    Button,
+    ButtonGroup,
+    Content,
+    Dialog,
+    DialogContainer,
+    Divider,
+    Flex,
+    Form,
+    Heading,
+    Item,
+    ListBox,
+    TextArea,
+    View,
+} from '@adobe/react-spectrum';
+import './share-visualization.scss';
+import { RouteComponentProps } from 'react-router-dom';
+import { IRootState } from 'app/shared/reducers';
+import { connect } from 'react-redux';
+
+import { getEntity as getVisualmetadataEntity } from 'app/entities/visualmetadata/visualmetadata.reducer';
+import { getEntity as getViewEntity } from 'app/entities/views/views.reducer';
+import { forwardCall } from 'app/shared/websocket/proxy-websocket.service';
+import { receiveSocketResponse, receiveSocketResponseByVisualId } from 'app/shared/websocket/websocket.reducer';
+import { renderVisualization, ValidateFields } from 'app/modules/canvas/visualization/util/visualization-render-utils';
+import { VisualWrap } from 'app/modules/canvas/visualization/util/visualmetadata-wrapper';
+import { getConditionExpression } from 'app/modules/canvas/filter/filter-util';
+import HeaderIcon from 'app/shared/components/header-icon/header-icon';
+import { translate } from 'react-jhipster';
+import Filter from '@spectrum-icons/workflow/Filter';
+import { toggleFilterPanel } from 'app/modules/canvas/filter/filter.reducer';
+import FilterPanel from 'app/modules/canvas/filter/filter-panel';
+import { setIsShare } from './share-visualization.reducer';
+
+export interface IShareVisualizationProps extends StateProps, DispatchProps, RouteComponentProps { }
+
+const ShareVisualization = (props: IShareVisualizationProps) => {
+
+    const params = new URLSearchParams(props.location.search);
+    const visualizationId = params.get('visualizationId');
+    const viewId = params.get('viewId');
+    const datasourceId = Number(params.get('datasourceId'));
+
+    useEffect(() => {
+        props.setIsShare(true);
+        if (visualizationId) {
+            props.receiveSocketResponse();
+            props.getVisualmetadataEntity(visualizationId);
+            props.getViewEntity(viewId);
+        }
+    }, []);
+
+    useEffect(() => {
+        if (props.visualmetadataEntity.fields && ValidateFields(props.visualmetadataEntity.fields)) {
```

Comment:
why you need this in useEffect..can you think a solun to avoid inside useEffect

Reply:
this useEffect is needed. its call when visualmetadataEntity receiver based on viz id then its call socket for data

Reply:
this might call more than once when the fields changes..use one boolean flag in reducer success function instead